app/controllers/points_controller.rb

```diff
@@ -27,98 +26,78 @@ def new
 
   def create
     @point = Point.new(point_params)
+    @cases = Case.includes(:topic).all
+
     @point.user = current_user
-    if params[:has_case]
-      if @point.case.nil? || @point.status.blank? || @point.source.blank?
-        flash[:alert] = "Oops! If you use a case, make sure that all the form fields are filled in before submitting!"
-        render :new
-      elsif
-        @point.update(title: @point.case.title, rating: @point.case.score, analysis: @point.case.description || @point.case.title, topic_id: @point.case.topic_id)
-        if @point.save
-          redirect_to service_path(@point.service)
-          flash[:notice] = "You created a point!"
-        else
-          render :new
-        end
-      end
-    elsif params[:only_create]
+
+    if params[:only_create]
       if @point.save
+        @point.topic_id = @point.case.topic_id
```

Comment:
Overall the code looks good, just one question about this: are we duplicating the topic of a case on the topic of a point, always? If so, is it not better to simply remove the connection of points to topics, and simply figure that out from a point's case?

Reply:
I was under the impression that we didn't want to have points floating around without topics. I agree that it's not the best idea to duplicate. I can remove the requirement that points have a topic, but in the longer term we may need to remove the topic_id field on points altogether.

Reply:
Yes, that was what I was suggesting :) I agree though that as long as we do have a `topic_id` field on points, it's fine to require it.